src/ripple/core/impl/JobQueue.cpp

```diff
@@ -150,98 +134,83 @@ JobQueue::getJobCountGE(JobType t) const
     // return the number of jobs at this priority level or greater
     int ret = 0;
 
-    std::lock_guard lock(m_mutex);
+    std::lock_guard lock(mutex_);
 
-    for (auto const& x : m_jobData)
+    for (auto const& jtd : jobData_)
     {
-        if (x.first >= t)
-            ret += x.second.waiting;
+        if (jtd.type() >= t)
+            ret += jtd.waiting;
     }
 
     return ret;
 }
 
-std::unique_ptr<LoadEvent>
-JobQueue::makeLoadEvent(JobType t, std::string const& name)
+LoadEvent
+JobQueue::createLoadEvent(JobType t, std::string name)
 {
-    JobDataMap::iterator iter(m_jobData.find(t));
-    assert(iter != m_jobData.end());
-
-    if (iter == m_jobData.end())
-        return {};
-
-    return std::make_unique<LoadEvent>(iter->second.load(), name, true);
+    return {jobData_[t].load().sample(), std::move(name), true};
```

Comment:
I spent some more time spelunking through the code today.  As far as I can tell it's okay that there is no lock here.  Sorry for the false alarm.  My instinct tells me that it would be good to leave a comment for why that lock is not needed.  But that's not essential.

In my defense, I think it's hard to review code using a type that requires locking only some of the time.  But I also know that part of the design predates your modifications, @nbougalis.